governance/program/src/processor/process_deposit_governing_tokens.rs

```diff
@@ -63,15 +68,35 @@ pub fn process_deposit_governing_tokens(
 
     realm_config_data.assert_can_deposit_governing_token(&realm_data, &governing_token_mint)?;
 
+    
     if is_spl_token_account(governing_token_source_info) {
         // If the source is spl-token token account then transfer tokens from it
-        transfer_spl_tokens(
-            governing_token_source_info,
-            governing_token_holding_info,
-            governing_token_source_authority_info,
-            amount,
-            spl_token_info,
-        )?;
+        match expected_mint_info {
+            Some(mint_info) => {
+                let additional_accounts = account_info_iter.as_slice();
```

Comment:
mprove clarity: would move this up and put it next to Some(expected_mint_info) in l.49

Reply:
the expected_mint_info is only used if the `governing_token_source_info` is a token_account.

in that case, for token extensions, it is needed for additional_accounts in transfer hooks and transfer fees. that is the reason why it is used in line 74.